### ISSUE TITLE
Failed recurring job hops from a designated queue to the default queue.

### ISSUE THREAD

**Issue by akhakpouri** [HangfireIO/Hangfire#2068]
This issue is very similar to #705!
I currently have a queue name `mx-cron`. I have a recurring task that runs every 6 hours. The job is created in my scheduler.

`RecurringJob.AddOrUpdate<IAccountJob>(nameof(AccountJob),
                job => job.Run(JobCancellationToken.Null),
                $"0 */6 * * *", TimeZoneInfo.Utc, queue: "mx-cron");`

Everything works fine if the job is completed successfully. However, when an exception is thrown, the retry is then queued in the "**default**" queue. I don't have a default queue created in my application; thus, the retries never happen.

The screenshot below describes this issue very well.

1. Job is created.
2. It's enqueued in the correct queue, **mx-cron**.
3. Exception is thrown. Job fails.
4. It is scheduled.
5. The scheduled retry is queued in the **default** queue.

![image](https://user-images.githubusercontent.com/4137746/182038942-b40197d9-6bcd-4a65-9721-9ad25f55bfbe.png)

I don't want the retry to hop from it's original queue.

**Comment by akhakpouri** [comment]
Hi - Any update on this issue?

**Comment by odinserj** [comment]
Unfortunately that `queue` parameter only works once, when background job is created, and doesn't work for its retries due to internal reasons. Version 1.8.0 will bring better support for queues that will solve this issue (and even deprecate this method that doesn't lead to predictable results). But in current versions the best solution for this is to use the `Queue` attribute and the following syntax to have queue consistent no matter what.

```csharp
[Queue("{0}")]
public static void MyMethod(string queue, int otherArg) {}
```

```csharp
BackgroundJob.Enqueue(() => MyMethod("mx-cron", 12345));
```

**Comment by samirvarandas** [comment]
Hello @odinserj, can you please confirm that this is solved in 1.8.0? Thanks :)

**Comment by odinserj** [comment]
Yes, can confirm it's closed when using new method overloads with queue specified explicitly.
